packages/fxa-settings/src/components/App/gql.ts

```diff
@@ -0,0 +1,55 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { gql } from '@apollo/client';
+
+// Glean init needs 'metricsEnabled' and 'uid'.
+// "Amplitude" init needs 'metricsEnabled', 'uid', 'recoveryKey',
+// if secondary email is verified, and `totp.verified`.
+export const INITIAL_METRICS_QUERY = gql`
+  query GetInitialMetricsState {
+    account {
+      uid
+      recoveryKey
+      metricsEnabled
+      emails {
+        email
+        isPrimary
+        verified
+      }
+      totp {
+        exists
+        verified
+      }
+    }
+  }
+`;
+
+/**
+ * This query reads from the local apollo-client cache and does not perform
+ * a network request and therefore is low-cost. It updates if any GQL network
+ * request response comes back with an 'invalid token' + 'unauthenticated' status
+ * (to `false`), and if a network request that requires auth succeeds. Because
+ * we have an initial metrics query that runs at the top of our app that requires
+ * a valid session token, this value will be updated on initial app start.
+ *
+ * If you create a new GQL query that should update this status to 'true', e.g.
+ * a query requiring a session token or a query that logs the user in, update
+ * the list in `lib/gql.ts` to include it.
+ *
+ * NOTE: This status can mean that the user has a valid session token, but
+ * does NOT currently tell you if that session is verified or unverified. We
+ * may be able to tell this in the GQL response as errno 138 from auth-server
+ * means "Unverified session", but we don't appear to currently need a verified
+ * session for GQL calls in non-Settings routes. In Settings, we're currently
+ * using auth-client for those calls.
+ *
+ * If you need to confirm that the session is _verified_, use
+ * `GET_SESSION_VERIFIED`. We can improve or revisit this in FXA-7626 or FXA-7184.
+ * */
+export const GET_LOCAL_SIGNED_IN_STATUS = gql`
+  query GetLocalSignedInStatus {
```

Comment:
Not exactly sure where to ask this, so I'll just ask here. What happens if this query is invoked, but the cache hasn't yet been primed?

Reply:
`isSignedIn` would come back as `undefined` in this case. This is accounted for in `App/index.tsx` like so:

```
if (loading || isSignedIn === undefined) {
```

Reply:
Great, I mostly wanted to double check it didn't totally blow up in the event there was a cache miss. I know that as the code stands today this shouldn't ever happen, but you never you know what can happen in the future.